scripts/e2e-test.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env sh
+
```

Comment:
```suggestion
set -e
```
I would recommend adding this stanza. And later run tests with if, e.g.:
```
if ! go test; then
  cleanup_fail=true
fi
```

It seems to me that this is a wrapper that will have tests added later one by one and people tend to expect failures not to be silent. And without `set -e` failures will be silent and will produce hard-to-debug time consuming problems when changes are added to the script. Even now the `cd` commands might fail if tests are moved but the error will be reported for `go test` line with error that will probably look like:

```
$ go test
go: cannot find main module; see 'go help modules'                                                                                                                                                                                        
```
which is confusing and doesn't point to the root cause.

You might wanna take a look at https://www.shellcheck.net/ that will help with quoting and similar issues.

Reply:
fixed 👍